presto-thrift-interface/src/main/java/com/facebook/presto/thrift/interfaces/readers/ByteColumnReader.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.thrift.interfaces.readers;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.ByteArrayBlock;
+import com.facebook.presto.thrift.interfaces.client.ThriftColumnData;
+
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+final class ByteColumnReader
+{
+    private ByteColumnReader()
+    {
+    }
+
+    private static void checkConsistency(boolean[] nulls, byte[] bytes, int totalRecords)
+    {
+        checkArgument(totalRecords == 0 || nulls != null || bytes != null, "nulls array or values array must be present");
+        checkArgument(nulls == null || nulls.length == totalRecords, "nulls must be null or of the expected size");
+        checkArgument(bytes == null || bytes.length == totalRecords, "bytes must be null or of the expected size");
+    }
+
+    public static Block readBlock(List<ThriftColumnData> columnsData, String columnName, int totalRecords)
+    {
+        ThriftColumnData columnData = ReaderUtils.columnByName(columnsData, columnName);
```

Comment:
What do you mean 'well-known'. I bought the last argument because the method name conflicted with another in the same class. This method does not suffer from the problem - there is only one implementation.